### PR TITLE
Fix rest eds default

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Small update to ensure rest EDS is enabled by default in environments that run off of k8s, add some unit tests to catch previous error.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by kdorosh** [comment]
/skip-changelog fixed the existing one instead of adding a new one

**Review by npolshakova — APPROVED** [review]
Looks good! I think you just need to skip the changelog.

File: test/services/gateway.go [hunk 1]
```diff
@@ -129,7 +135,21 @@ func RunGlooGatewayUdsFds(ctx context.Context, runOptions *RunOptions) TestClien
...
+	} else {
+		if glooOpts.Settings.Gloo == nil {
+			glooOpts.Settings.Gloo = &gloov1.GlooOptions{
+				RestXdsBindAddr: fmt.Sprintf("0.0.0.0:%v", int(runOptions.RestXdsPort)),
+			}
+		} else {
+			if glooOpts.Settings.Gloo.RestXdsBindAddr == "" {
+				glooOpts.Settings.Gloo.RestXdsBindAddr = fmt.Sprintf("0.0.0.0:%v", int(runOptions.RestXdsPort))
+			}
+		}
```
Comment (kdorosh) [inline test/services/gateway.go:152]: we should be able to DRY this considerably, i.e.
```golang
if glooOpts.Settings == nil {
    glooOpts.Settings = &gloov1.Settings{}
}
if glooOpts.Settings.Gloo == nil {
    glooOpts.Settings.Gloo = &gloov1.GlooOptions{}
}
if glooOpts.Settings.Gloo.RestXdsBindAddr == "" {
    glooOpts.Settings.Gloo.RestXdsBindAddr = fmt.Sprintf("0.0.0.0:%v", int(runOptions.RestXdsPort))
}
```

**Comment by ashleywang1** [comment]
Note that the aws_ec2 test failed once, but I ran it with gcloud build and it passed 30 times with a 20 second timeout. If it occurs again, increase the timeout from 10s to 20s.

**Review by kdorosh — COMMENTED** [review]
lgtm! can't approve because it's technically my PR

**Review by npolshakova — APPROVED** [review]

**Pull request merged**